### ISSUE TITLE
MixMatch: A Holistic Approach to Semi-Supervised Learning (NeurIPS 2019)

### ISSUE THREAD

**Issue by iBotamon** [yutanakamura-tky/paper_reading#4]
https://papers.nips.cc/paper/8749-mixmatch-a-holistic-approach-to-semi-supervised-learning.pdf

# どんなもの？
    - 半教師あり学習(Semi-supervised Learning (SSL))において主流の複数のアプローチを1つにまとめたMixMatchという手法を提案した研究.
    - ライブラリもある -> https://github.com/google-research/mixmatch
# 先行研究と比べてどこがすごい？
    - 実質, 既存の手法がぜんぶ盛りになっている
    - 以下は半教師あり学習の既存のアプローチ. MixMatchは以下の3つの要素をすべて取り入れている.
        - Consistency Regularization
            - クラスの決定と関係のない素性について一般化する
            - いわゆる画像のAugmentationと同義(例:平行移動することによって絶対的な位置について一般化する)
            - 具体的には...
                - 分類問題を考え, ラベルのついていないデータ点をxとする
                - xにAugmentationを施したデータ点をx1,x2とする
                - x1,x2は同じクラスに分類されて欲しい
                - このために例えば次のようなことを行う
                    - Ref.25,40: Loss関数にp(y|x1;θ)とp(y|x2;θ)の2乗ノルムを加える
                    - Ref.44: 上記のp(y|x1;θ)をp(y|x1;θ+Δθ)に変える. Δθはパラメーターのexponential moving average
            - この方法の欠点はAugmentationにモダリティに特有のアプローチが必要であること(画像なら平行移動/反転/回転,など)
            - ただしVirtual Adversarial Training (Ref.31) は摂動を用いてモダリティによらないAugmentationを提唱している
            - MixMatchで使用するのはモダリティに特有のアプローチ
        - Entropy Minimization
            - SSLは決定境界が周辺確率のhigh-density regionsを通過しないことを要請している
            - そのためには, ラベルのついていないデータ点に対し, モデルにできるだけエントロピーの低い出力をさせればよい
            - 方法はいくつかある
                - エントロピー自体をLoss関数を加えるもの(Ref.18)
                - Pseudo-Label(Ref.28)も実はこの部類に入る
        - Traditional Regularization
            - つまりは正則化. モデルの複雑さが上がることに対してペナルティを付加する.
# 技術や手法のキモはどこ？
    - 以下, ラベルの"付いた"データ点をx, 付いていないデータ点をuとする.
    - 1. Augmentation: 各x, uに対してK回augmentationを行い, x^, u1,...,uKを得る
    - 2. Label Guessing: u1,...,uKに対するモデルの予測の平均をPseudo-Labelとする (q=mean(p(y|uk;θ)))
    - 3. Sharpening: qの各クラスに対する事後確率をシャープにする
        - Lクラス分類であり, q = (q1,...,qL) とおくと, q -> {qi**(1/T) / sum(qc**(1/T) for c in range(L))}
        - Tは温度を模したハイパラ. T -> +0 の極限をとると q はone-hot vectorに, qのエントロピーは+0に近づく
    - 4. MixUp: (データ点,事後確率)の組を2つとり, 線形補完して新しいデータ点をつくる
        - αをハイパラとする
        - λをBeta分布(α,α)からサンプルし, 関数MixUp(r1,r2)をr1,r2の(1-λ):λ内分点を返す関数とおく
        - X={(x^, p(y|x^;θ))}, U={(uk, p(y|uk;θ))} とおき, XとUの和集合をシャッフルしたものをWとおく
        - X, U内の各データ点に対し,MixUp相手をWから非復元抽出によってえらび,MixUpを計算する. するともとのデータ点と同数のデータ点が新しく生成される. 新しいデータ点とX, Uの和集合をX', U'とする
        - まとめると, MixUpの過程でX, U から X', U'が生成され, データ量は2倍になっている
    - 5. Loss Function
        - X'に対するLoss: Lx = mean(cross-entropy(p'(y|x';θ), p_model(y|x';θ))) for x', p' in X'
            - X'の事後確率にはモデルが出力したものと線形補完で計算されたものが混在している
            - よって, 線形補完されているものについて, 本来モデルが出力するはずの確率とのcross-entropyを求め, 平均をとる
        - U'に対するLoss: Lu = 1/クラス数 * mean(L2Norm(q', p_model(y|u';θ)) for u', q' in U'
            - Pseudo-labelを教師ラベルとした最小二乗法
        - 全体: L = Lx + λuLu (λuはハイパラ)
# どうやって有効だと検証した？
    - ハイパラ
        - T=0.5, K=2, α=0.75, λu=100とした
    - 画像分類データセット(CYFAR-10, SVHN, STL-10)で検証, 既存のどの半教師あり学習よりも高性能だった
# 議論はある？
    - MixMatchを用いるとモデルのプライバシーリスクが減る(なぜなら少数のデータでもoverfitしないため)
    - 差分プライバシーのε値によって定量評価したところ,既存の半教師あり学習であるVATとくらべε=4.96->0.97と大幅に改善
# 次に読むべき論文は？
